### PR TITLE
fix to spit out time correctly, not DateTime object

### PR DESCRIPTION
**Pull request by siddwarkhedkar**

### CODE REVIEW

File: src/Model/EventModel.php [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 use Klaviyo\Exception\KlaviyoException;
 use Klaviyo\Model\ProfileModel;
+use DateTime;
```
Comment (smoucka) [inline src/Model/EventModel.php:7]: nit: let's do something similar to our app code organization and import standard classes above the package classes

Comment (smoucka) [reply src/Model/EventModel.php:7]: can separate with a newline too if you want

File: src/Model/EventModel.php [hunk 2]
```diff
@@ -40,10 +41,11 @@ public function __construct(array $config ) {
...
         // Can pass in unix timestamp if prefixed with '@'. Else just let it parse the date
-        $this->time = !empty($config['time']) ?
-            new DateTime(
-                is_int($config['time']) ? '@' . $config['time'] : $config['time']
-            ) : null;
+        $time = (array) new DateTime(
+            is_int($config['time']) ? '@' . $config['time'] : $config['time']
+        );
+
+        $this->time = !empty($config['time']) ? $time['date'] : null;
```
Comment (smoucka) [inline src/Model/EventModel.php:48]: I could be wrong but pretty sure we need to convert this to a unix timestamp in order to successfully send to the track api. we can either do this right here or can we make that conversion in the jsonSerialize method

File: src/Model/EventModel.php [hunk 3]
```diff
@@ -45,7 +45,7 @@ public function __construct(array $config ) {
             is_int($config['time']) ? '@' . $config['time'] : $config['time']
         );
 
-        $this->time = !empty($config['time']) ? $time['date'] : null;
+        $this->time = !empty($config['time']) ? strtotime( $time['date'] ) : null;
```
Comment (smoucka) [inline src/Model/EventModel.php:49]: nice

File: src/Model/EventModel.php [hunk 4]
```diff
@@ -40,10 +42,11 @@ public function __construct(array $config ) {
         );
         $this->properties = $config['properties'];
         // Can pass in unix timestamp if prefixed with '@'. Else just let it parse the date
-        $this->time = !empty($config['time']) ?
-            new DateTime(
-                is_int($config['time']) ? '@' . $config['time'] : $config['time']
-            ) : null;
+        $time = (array) new DateTime(
```
Comment (smoucka) [inline src/Model/EventModel.php:45]: DateTime parses a ton of different date formats but it's definitely possible someone will pass in something invalid. We should probably catch any exceptions and throw a klaviyo exception

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,8 @@
 ## CHANGELOG
 
+### 2.1.1
+- Fix - EventModel configuration to use DateTime correctly
```
Comment (smoucka) [inline CHANGELOG.md:4]: Can you please add a line here to address the profile model change too?

File: src/Model/EventModel.php [hunk 6]
```diff
@@ -39,11 +41,16 @@ public function __construct(array $config ) {
             $config['customer_properties']
         );
         $this->properties = $config['properties'];
-        // Can pass in unix timestamp if prefixed with '@'. Else just let it parse the date
```
Comment (smoucka) [inline src/Model/EventModel.php:42]: Can you update the docstring for this param to mention that it accepts either a unix timestamp OR any date format that DateTime accepts?

File: src/Model/EventModel.php [hunk 7]
```diff
@@ -39,11 +41,16 @@ public function __construct(array $config ) {
...
-        $this->time = !empty($config['time']) ?
-            new DateTime(
+
+        try {
+            $time = (array)new DateTime(
                 is_int($config['time']) ? '@' . $config['time'] : $config['time']
-            ) : null;
+            );
+        } catch ( Exception $e ) {
```
Comment (smoucka) [inline src/Model/EventModel.php:49]: You're going to need to import Exception as well

**Comment by siddwarkhedkar** [comment]
@smoucka @remstone7 I guess we're good to go here?

**Pull request merged**